### DISCUSSION TITLE
Is it possible to define a diff-filter for conditions?

### DISCUSSION THREAD

**Discussion by alfredbez** [captainhook-git/captainhook#208] [General]
Would be nice to be able to define the diff-filter for `FileChanged` conditions, maybe like this:
```json
{
  "action": "console my:command",
  "options": {},
  "conditions": [
    {
      "exec": "\\CaptainHook\\App\\Hook\\Condition\\FileChanged\\Any",
      "args": [
        [
          "src/*/*Foo.php",
          "src/*/*Bar.php",
          "src/*/*Baz.php",
        ],
        "ACDR"
      ]
    }
  ]
}
```

I have this scenario: There is a pretty heavy console command, that collects some classes in a map. This map just contains the paths to the files (pretty similar to the class map in composer), it does not really care about what's inside these files. In most of the cases we're changing something in these files, but we rarely add new files or remove existing ones, but the action is triggered each time if a matching file is changed.

Do you think something like this would be possible? Or should I rather create my own condition for that?

**Comment by sebastianfeldmann** [comment]
Now with version `5.16.2` you can, exactly like you proposed :)
